### PR TITLE
BUG: fix typos in constants

### LINKED ISSUE #205: SI prefix milli is misspelled
* unyt version: 2.8.0
* Python version: 3.7.6
* Operating System: Windows

### Description
The name of the SI prefix milli appears to be misspelled, with only a single l, as "mili".  One place (maybe the only place) where this occurs in the code is https://github.com/yt-project/unyt/blob/e86c0500e9c9f29914eed3ef7d8f11c98768eedd/unyt/_unit_lookup_table.py#L325 .

### What I Did
```
>>> import unyt
>>> unyt.milimeter
mm
>>> unyt.millimeter
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'unyt' has no attribute 'millimeter'
```

### PR DESCRIPTION
**Pull request by neutrinoceros**
fix #205

### CODE REVIEW

**Comment by neutrinoceros** [comment]
also fixes #222
Switching to draft while I try to think about a nicer way to fix these without breaking existing user code. Of course a simple way to do that would be to simply have both the correct and the historic-and-mispelled versions coexist for a while. The hard part is making the erroneous version visibly deprecated.

**Comment by neutrinoceros** [comment]
also rebased on top of #219 to stabilise CI

**Comment by neutrinoceros** [comment]
I seems tricky to properly deprecate for these, so I've switched the PR to a more conservative and backward compatible approach for now

File: unyt/tests/test_unyt_array.py [hunk 1]
```diff
@@ -828,6 +832,23 @@ def test_ytarray_pickle():
         assert_equal(float(data.units.base_value), float(loaded_data.units.base_value))
 
 
+@pytest.mark.xfail(
```
Comment (jzuhone) [inline unyt/tests/test_unyt_array.py:835]: @neutrinoceros is this related to the rest of the PR?

Comment (jzuhone) [reply unyt/tests/test_unyt_array.py:835]: Never mind, I saw you had to rebase on #219. We'll get this sorted soon.

**Review by jzuhone — APPROVED** [review]
